CI issue - NPM fails to install SF CLI

NPM is failing to install SF CLI and due to that - CI keeps failing to run PR checks as well as automated builds. 

This can be fixed by manually building the package but it has to be fixed

```bash
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: salesforce-app@1.0.0
npm error Found: @lwc/eslint-plugin-lwc@1.9.0
npm error node_modules/@lwc/eslint-plugin-lwc
npm error   dev @lwc/eslint-plugin-lwc@"^1.1.2" from the root project
npm error
npm error Could not resolve dependency:
npm error peer @lwc/eslint-plugin-lwc@"^2.0.0" from @salesforce/eslint-config-lwc@3.7.1
npm error node_modules/@salesforce/eslint-config-lwc
npm error   dev @salesforce/eslint-config-lwc@"^3.2.3" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /home/runner/.npm/_logs/202[5](https://github.com/kryvyifedir/score-hub/actions/runs/13774721254/job/38521309200#step:3:6)-03-10T21_03_26_980Z-eresolve-report.txt
npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2025-03-10T21_03_2[6](https://github.com/kryvyifedir/score-hub/actions/runs/13774721254/job/38521309200#step:3:7)_980Z-debug-0.log
Error: Process completed with exit code 1.
```

--- a/.github/workflows/pullrequest_build.yml
+++ b/.github/workflows/pullrequest_build.yml
@@ -15,9 +15,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       # Checkout the Source code from the latest commit
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
+          ref: ${{ github.event.pull_request.head.ref }}
           fetch-depth: 0
+
+      - name: Verify branch
+        run: git branch 
 
       # Installing NPM so that SFDX can be installed
       - name: Installing NPM

--- a/.github/workflows/pullrequest_release.yml
+++ b/.github/workflows/pullrequest_release.yml
@@ -15,7 +15,11 @@ jobs:
       # Checkout the Source code from the latest commit
       - uses: actions/checkout@v4
         with:
+          ref: dev
           fetch-depth: 0
+          
+      - name: Verify branch
+        run: git branch 
     
       # Installing NPM so that SFDX can be installed
       - name: Installing NPM

